conda/plugins/manager.py

```diff
@@ -9,24 +9,34 @@
 from . import solvers, virtual_packages
 from .hookspec import CondaSpecs, spec_name
 from ..auxlib.ish import dals
-from ..exceptions import PluginError
+from ..base.context import context
+from ..core.solve import Solver
+from ..exceptions import CondaValueError, PluginError
 
 
 class CondaPluginManager(pluggy.PluginManager):
     """
     The conda plugin manager to implement behavior additional to
     pluggy's default plugin manager.
     """
+    #: Cached version of the :meth:`~conda.plugins.manager.CondaPluginManager.get_solver_backend`
+    #: method.
+    get_cached_solver_backend = None
 
-    def __init__(self, project_name: str | None = None, *args, **kwargs):
+    def __init__(self, project_name: str | None = None, *args, **kwargs) -> None:
         # Setting the default project name to the spec name for ease of use
         if project_name is None:
             project_name = spec_name
         super().__init__(project_name, *args, **kwargs)
+        # Make the cache containers local to the instances so that the
+        # reference from cache to the instance gets garbage collected with the instance
```

Comment:
https://www.youtube.com/watch?v=sVjtp6tGo0g is a good summary of this problem. Or https://rednafi.github.io/reflections/dont-wrap-instance-methods-with-functoolslru_cache-decorator-in-python.html.